c7n/resources/glue.py

```diff
@@ -520,6 +520,8 @@ def get_model(self):
     def _get_catalog_encryption_settings(self):
         client = utils.local_session(self.session_factory).client('glue')
         settings = client.get_data_catalog_encryption_settings()
+        if not settings.get('CatalogId'):
+            settings['CatalogId'] = self.config.account_id
```

Comment:
Based off of the docs, this seems fine. Interesting behavior though.

Reply:
this shouldn't be conditional though, catalogid is not a returned api attribute

Reply:
Thanks, I have removed the `CatalogId` check! Had taken it into consideration just in case aws starts shipping that as part of response. But I agree, its better to remove it.